arkane/ess/psi4.py

```diff
@@ -0,0 +1,309 @@
+#                                                                             #
+# Permission is hereby granted, free of charge, to any person obtaining a     #
+# copy of this software and associated documentation files (the 'Software'),  #
+# to deal in the Software without restriction, including without limitation   #
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,    #
+# and/or sell copies of the Software, and to permit persons to whom the       #
+# Software is furnished to do so, subject to the following conditions:        #
+#                                                                             #
+# The above copyright notice and this permission notice shall be included in  #
+# all copies or substantial portions of the Software.                         #
+#                                                                             #
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,    #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+Arkane Psi4 module
+Used to parse Psi4 output files
+"""
+
+import logging
+
+import numpy as np
+
+import rmgpy.constants as constants
+from rmgpy.statmech import IdealGasTranslation, NonlinearRotor, LinearRotor, HarmonicOscillator, Conformer
+
+from arkane.common import  get_element_mass, get_principal_moments_of_inertia, convert_imaginary_freq_to_negative_float
+from arkane.exceptions import LogError
+from arkane.ess.adapter import ESSAdapter
+from arkane.ess.factory import register_ess_adapter
+
+################################################################################
+
+
+class Psi4Log(ESSAdapter):
+    """
+    Represent an output file from Psi4. The attribute ``path`` refers to the
+    location on disk of the Psi4 output file of interest. Methods are provided
+    to extract a variety of information into Arkane classes and/or NumPy
+    arrays. Psi4Log is an adapter for the abstract class ESSAdapter.
+    """
+
+    def check_for_errors(self):
+        """
+        Checks for common errors in a Gaussian log file.
+        If any are found, this method will raise an error and crash.
+        """
+        with open(self.path) as f:
+            log = f.readlines()
+        for line in reversed(log):
+            if 'Psi4 exiting successfully' in line:
+                break
+        else:
```

Comment:
After checking for normal completion, can we add an elif to look for some specific errors and then print that line for user convenience? For example, it seems that keywords indicating errors include `RuntimeError`, `Fatal Error`, and/or `PSIO Error`. Can you add a unit test for the `check_for_errors` method and perhaps parse the failed output file from this discussion board ([link](http://forum.psicode.org/t/psio-error-during-adc-2/1338))? Feel free to add any failed runs from your work.

Reply:
Excellent suggestion. I added the conditions and the online example. I don't have experience with failed Psi4 runs. @kfir4444, would you know of typical Psi4 fails and corresponding phrases in output files?